### PR TITLE
Fixes #3931: allow Kohana shutdown properly, exit(1) in case of exception

### PR DESCRIPTION
**Pull request by enov**

### CODE REVIEW

**Comment by zeelot** [comment]
This looks ok to me… seemed to work. Will not merge this into 3.3+ since we have a different fix in there that involves a better API.

**Comment by enov** [comment]
Hey @Zeelot , would you mind also closing #4275 and #4794 on Redmine? They depended on this bug and this PR would fix them too.

**Comment by zeelot** [comment]
Done, thanks.

File: classes/kohana/core.php [hunk 1]
```diff
@@ -967,11 +967,24 @@ public static function error_handler($code, $error, $file = NULL, $line = NULL)
...
+	 * on the second visit.
+	 * @see issue #3931
 	 *
 	 * @return  void
 	 */
 	public static function shutdown_handler()
 	{
+		// test if we have already shutdown, if TRUE, exit(1)
+		static $shutdown_handler_visited = FALSE;
+		if ($shutdown_handler_visited) {
```
Comment (evanpurkhiser) [inline classes/kohana/core.php:983]: Not to be too picky, but the brace here should be on the next line.

Comment (enov) [inline classes/kohana/core.php:983]: @EvanPurkhiser it seems I need to PR anew?

Comment (zeelot) [inline classes/kohana/core.php:983]: yeah if you want to :) I don't actually care all that much.

Comment (enov) [inline classes/kohana/core.php:983]: @Zeelot: voilà https://github.com/kohana/core/pull/463

**Pull request merged**